lib/modelinstance.js

```diff
@@ -95,24 +96,30 @@ ModelInstance.applyData = function (mdlInst, data) {
  * attempts to save it to the data store.
  *
  * @param {Object} data
- * @param {Function} callback
+ * @param {Function=} callback
+ * @returns Promise
  */
 ModelInstance.create = function (data, callback) {
-  var ctorArgs = [''];
-  for (var i = 0; i < arguments.length - 1; ++i) {
-    ctorArgs.push(arguments[i]);
-  }
-
-  var mdlInst = // jshint -W058
-    new (Function.prototype.bind.apply(this, ctorArgs));
+  var self = this;
+  var args = arguments;
 
-  mdlInst.save(function (err) {
-    if (err) {
-      callback(err, null);
-      return;
+  return new Promise(function (resolve, reject) {
+    var ctorArgs = [''];
+    for (var i = 0; i < args.length && !(args[i] instanceof Function); ++i) {
+      ctorArgs.push(args[i]);
     }
-    callback(null, mdlInst);
-  });
+
+    var mdlInst = // jshint -W058
+      new (Function.prototype.bind.apply(self, ctorArgs));
+
+    mdlInst.save(function (err) {
+      if (err) {
+        reject(err);
```

Comment:
Suggest common idiom here `return reject(err);`.   It saves the second line of code (return) and it guarantees that when the outer promise returns or rejects, you never fall through in the code and keep going, which is a risk.